nedap/formatting-stack
# fix test-runner/test!

## Brief

`#'git-completely-staged` and `#'git-not-completely-staged` both require `:files` to be a collection since c131026f5391978350ecf32eaddddf37c6785e71

The test is non-ideal as it's testing a side-effect of `test!` not crashing on the calls to the strategies. Any ideas are welcome.


## QA plan

(copied from https://github.com/nedap/formatting-stack/pull/79)

* install `formatting-stack 4.1.2-alpha1`
* ensure `"test"` is included in `(set-refresh-dirs ...)` 
* `lein with-profile +test repl`
* `(require '[formatting-stack.compilers.test-runner :refer [test!]])`
* make sure to run `(refresh)`

#### Feature 1

* Touch an implementation ns which is covered by a _testing ns_, so `git diff` will see it.
* run `(test!)`
* The tests of that ns (which `git diff` was _not_ necessarily seeing) will be run.

#### Feature 2

* Touch a testing ns, so `git diff` will see it.
* run `(test!)`
* That test will be run.

#### Feature 3

* Repeat 1 or 2 but `git commit`ting first, in a branch other than master.
  * You will need `(test! :target-branch "staging")` if your default branch isn't `master`.
* The same namespaces will be run as before: now the filenames were derived out of the "branch diff" instead of `git status`.

## Author checklist



* Additionally, I have code-reviewed iteratively the PR considering the following aspects in isolation:

## Reviewer checklist

* Additionally, I have code-reviewed iteratively the PR considering the following aspects in isolation:
